src/Common/PoolWithFailoverBase.h

```diff
@@ -249,51 +249,44 @@ PoolWithFailoverBase<TNestedPool>::getMany(
     });
 
     std::string fail_messages;
-    bool finished = false;
-    while (!finished)
```

Comment:
Why? It may happen that we won't get enough connections during one loop, also we can try to establish the same connection several times (up to max_tries)